crates/cairo-lang-utils/src/bigint.rs

```diff
@@ -112,44 +112,39 @@ where
 
 impl Encode for BigIntAsHex {
     fn size_hint(&self) -> usize {
-        // sign, len, data.
-        1 + 8 + self.value.to_bytes_be().1.len()
+        // sign, data.
+        core::mem::size_of::<u8>() + self.value.bits() as usize / 8
```

Comment:
YOu are missing one u8 there.
Sign takes one, vec len takes one, and then the rest is `self.value.bits() as usize / 8`

Reply:
s there a domain constraint we can rely upon in order to be sure that the vec won't be more than X byte long?

If those BigInt are used to store Felt252, we know for sure that the data len won't exceed 32. Which is totally encodable on a single byte with compact scale codec

Reply:
Is there a way to be sure that `self.value.bits() as usize / 8` is always equal to `self.value.to_bytes_le().len()`?
Can we have some unit tests for this?
Or at least a comment that says that it has been tested and is always the case. In a few month it will be hard to make the link between those two just by reading the code.

Reply:
It surely does not always equal as it's an integer div we need to ceil 🫠

Reply:
As for the total size as it's a hint we don't have any guarantees that it'll be a felt252 there (imo)

Reply:
We should rather round-up than down, shouldn't we?